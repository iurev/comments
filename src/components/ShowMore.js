import React, { Component } from 'react';
import styled, { css } from 'styled-components'
import { connect } from 'react-redux'
import Ink from 'react-ink'
import { ifProp } from 'styled-tools'
import { createSelector } from 'reselect'
import actions from '../actions'
import {
  feedbacksList,
  feedbacksLimit,
} from '../selectors'

const Root = styled.button`
  position: relative;
  border: 2px solid ${p => p.theme.green};
  width: 70px;
  height: 70px;
  border-radius: 35px;
  color: ${p => p.theme.green};
  display: flex;
  align-items: center;
  justify-content: center;
  background: transparent;
  font-weight: 700;
  font-size: 9px;
  cursor: pointer;
  outline: none;
  transition: background ease-out 0.3s;
  user-select: none;

  &:hover {
    background: ${p => p.theme.lightGreen};
  }

  ${ifProp('loading', css`
    border: 2px solid ${p => p.theme.mediumFontColor};
    color: ${p => p.theme.mediumFontColor};
    cursor: default;
    pointer-events: none;
    &:hover {
      background: transparent;
    }
  `)}
`

const selector = createSelector(
  feedbacksList,
  feedbacksLimit,
  (list, limit) => ({ hidden: list.length <= limit })
)

const ShowMore = ({ loading, showMore, hidden }) => {
  if (hidden) return null
  return (
    <Root
      onClick={showMore}
      disabled={loading}
      loading={loading}
    >
      <Ink />
      {loading ? 'Loading' : 'Load more'}
    </Root>
  );
}

export default connect(
  state => selector(state),
  actions,
)(ShowMore);
