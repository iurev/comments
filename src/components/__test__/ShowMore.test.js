import React from 'react'
import ShallowRenderer from 'react-test-renderer/shallow'
import { ShowMore } from '../ShowMore'


it('hidden', () => {
  const tree = new ShallowRenderer().render(<ShowMore hidden />)

  expect(tree).toMatchSnapshot()
})

it('loading', () => {
  const tree = new ShallowRenderer().render(<ShowMore loading />)

  expect(tree).toMatchSnapshot()
})

it('ok', () => {
  const tree = new ShallowRenderer().render(<ShowMore loading={false} />)

  expect(tree).toMatchSnapshot()
})
