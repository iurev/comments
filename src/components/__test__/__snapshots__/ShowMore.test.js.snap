// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`hidden 1`] = `null`;

exports[`hidden 2`] = `null`;

exports[`loading 1`] = `
<ForwardRef
  disabled={true}
  loading={true}
>
  <e
    background={true}
    className="ink"
    duration={1000}
    opacity={0.25}
    radius={150}
    recenter={true}
  />
  Loading
</ForwardRef>
`;

exports[`ok 1`] = `
<ForwardRef
  disabled={false}
  loading={false}
>
  <e
    background={true}
    className="ink"
    duration={1000}
    opacity={0.25}
    radius={150}
    recenter={true}
  />
  Load more
</ForwardRef>
`;
